pkg/disttask/framework/taskexecutor/execute/interface_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package execute
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSubtaskSummaryGetSpeed(t *testing.T) {
+	tests := []struct {
+		name        string
+		setup       func(*SubtaskSummary)
+		endTime     time.Time
+		duration    time.Duration
+		expected    int64
+		description string
+	}{
+		{
+			name: "insufficient data points",
+			setup: func(s *SubtaskSummary) {
+				s.Progresses = []Progress{
+					{Bytes: 100, UpdateTime: time.Unix(1000, 0)},
+				}
+			},
+			endTime:     time.Unix(1010, 0),
+			duration:    time.Second * 10,
+			expected:    0,
+			description: "should return 0 when less than 2 data points",
+		},
+		{
+			name: "no overlap with data range",
+			setup: func(s *SubtaskSummary) {
+				baseTime := time.Unix(1000, 0)
+				s.Progresses = []Progress{
+					{Bytes: 0, UpdateTime: baseTime},
+					{Bytes: 100, UpdateTime: baseTime.Add(1 * time.Second)},
+				}
+			},
+			endTime:     time.Unix(1010, 0),
+			duration:    time.Second, // [9, 10]
```

Comment:
```suggestion
			duration:    time.Second, // [1009, 1010]
```
please update below too